### PR TITLE
 Enable CA2249: Consider using String.Contains instead of IndexOf

### PR DESCRIPTION
**Pull request by xtqqczze**
https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2249

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs [hunk 1]
```diff
@@ -1985,7 +1985,7 @@ private static string EscapeFunctionNameForRemoteHelp(string name)
             StringBuilder result = new(name.Length);
             foreach (char c in name)
             {
-                if (("\"'`$".IndexOf(c) == (-1)) &&
+                if ((!"\"'`$".Contains(c)) &&
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs:1988]: Maybe:
```suggestion
                if (!("\"'`$".Contains(c)) &&
```

File: src/System.Management.Automation/cimSupport/cmdletization/ScriptWriter.cs [hunk 2]
```diff
@@ -832,7 +832,7 @@ private static string EscapeModuleNameForHelpComment(string name)
             StringBuilder result = new(name.Length);
             foreach (char c in name)
             {
-                if (("\"'`$#".IndexOf(c) == (-1)) &&
+                if ((!"\"'`$#".Contains(c)) &&
```
Comment (iSazonov) [inline src/System.Management.Automation/cimSupport/cmdletization/ScriptWriter.cs:835]: The same.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs (hunk 1 above)
Comment (xtqqczze) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs:1988]: The parenthesis are unnecessary and would be a rule violation of [IDE0047](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0047) and [SA1119](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1119.md).

Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs:1988]: Why not remove the parenthesis at all?
```suggestion
                if (!"\"'`$".Contains(c) &&
```

Comment (xtqqczze) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs]: I try to submit code quality PRs without style changes, but as you asked, I've pushed some more commits to address these issues.

**Review by iSazonov — APPROVED** [review]

**Pull request merged**